### PR TITLE
Autoinst Overhaul

### PR DESCRIPTION
**Pull request by Loquacity**
# Description

Rearranges all the Autoinst files in the Client Cfg, adds some more exposition, and incorporates content from the Ref Guide.

# Target branches

Which product version does this PR apply to?

- [ ] Manager-4.0
- [ ] Manager-4.1
- [x] Manager-4.2
- [x] Uyuni

# Links

Fixes https://github.com/SUSE/spacewalk/issues/12355

### CODE REVIEW

**Comment by Loquacity** [comment]
(lets see what xrefs break, then I'll generate the files to make it easier to review)

**Comment by Loquacity** [comment]
image
[uyuni_client-configuration_guide.pdf](https://github.com/uyuni-project/uyuni-docs/files/5192850/uyuni_client-configuration_guide.pdf)

File: modules/client-configuration/pages/autoinst-setup.adoc [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+An autoinstallation file is a method of providing this information in a scripted form.
+They are sometimes referred to as answers files.
+
+Within {productname}, you can use two different types of autoinstallation files, depending on the operating system of the clients you want to install.
+For {sle} clients, use {ay}.
+For {rhel} clients, use {kickstart}.
+You can use both {ay} and {kickstart} files if you want to install clients with different operating systems.
+
+* For information about creating {ay} autoinstallation files, including example files, see xref:client-configuration:autoyast.adoc[].
+* For information about creating {kickstart} autoinstallation files, see xref:client-configuration:autoyast.adoc[].
```
Comment (keichwa) [inline modules/client-configuration/pages/autoinst-setup.adoc:69]: Here we need a kickstart resource.  I think this is:
```suggestion
* For information about creating {kickstart} autoinstallation files, see xref:client-configuration:kickstart.adoc[].
```

**Review by keichwa — APPROVED** [review]

File: modules/client-configuration/pages/autoinst-profiles.adoc [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+== File Preservation for {kickstart}
+
+If you have many custom configuration files located on a client you want to {kickstart}, you can save them as a list, and associate that list with the {kickstart} profile to be used.
+
+In the {productname} {webui}, navigate to menu:Systems[Autoinstallation > File Preservation] and click btn:[Create File Preservation List].
+Enter a suitable label, and list absolute paths to all files and directories you want to save.
+Click btn:[Create List].
+
+Include the file preservation list in your {kickstart} profile by navigating to menu:Systems[Autoinstallation > Profiles] and selecting the profile you want to edit.
+Naviagte to the menu:System Details[File Preservation] subtab and select the file preservation list to include.
```
Comment (Loquacity) [inline modules/client-configuration/pages/autoinst-profiles.adoc:77]:
```suggestion
Navigate to the menu:System Details[File Preservation] subtab and select the file preservation list to include.
```

File: modules/client-configuration/pages/autoinst-setup.adoc [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+
+
+=== Variables
+
+Autoinstallation variables can be used to substitute values into Kickstart and {ay} profiles.
+To define a variable, create a name-value pair ([replaceable]``name/value``) in the text box.
+
+For example, if you want to autoinstall a system that joins the network of a specified organization (for example the Engineering department) you can create a profile variable to set the IP address and the gateway server address to a variable that any system using that profile will use.
+Add the following line to the [guimenu]``Variables`` text box.
```
Comment (Loquacity) [inline modules/client-configuration/pages/autoinst-setup.adoc:82]:
```suggestion
Add this line to the [guimenu]``Variables`` text box.
```

**Pull request merged**